internal_docs/technical-architecture.md

```diff
@@ -0,0 +1,111 @@
+
+We want prototypes to be hidden from search engines and hard for normal internet users to access, so we require that prototypes hosted online have a password. We also require HTTPS and change a few other things to improve the security of the prototype.
+
+### Manage prototype pages
+
+Some parts of the prototype can be configured or inspected via webpages at `http://localhost:3000/manage-prototype` (only when running locally). These web pages are created by a separate app, so that prototype authors are not able to modify these pages.
+
+### Plugins
+
+We want people to be able to extend the capabilities of the kit. We have a plugin subsystem that detects npm packages that contain a `govuk-prototype-kit-config.json` manifest and exposes files described by that manifest to the prototype. For instance, this can be used to share layouts and templates.
+
+## What the kit comes with
+
+To help users create prototypes, the kit has a number of dependencies of its own.
+
+These include:
+
+- [Node.js Express][expressjs] framework
+
+This provides the basic working of the kit - a web server that can handle requests, render templates to HTML and respond to the browser.
+
+- [Nunjucks][nunjucks] templating
+
+This allows us to keep each page template simple - and inherit the [GOV.UK Frontend][govuk-frontend] template from one place. It has powerful features such as looping over an array of variables, that make more complex prototypes much easier.
+
+- [Nodemon][nodemon] to watch files and restart the server
+
+This means any changes to routes.js or other server files are updated instantly in the prototype.
+
+- [Browsersync][browsersync] to reload the browser with any changes
+
+We serve pages in a user's prototype via Browsersync. This means any changes to user templates, Sass or JavaScript are updated instantly in the browser without needing to manually refresh.  Note that this does not work for the manage prototype pages.
+
+- [Portscanner][portscanner] to find a free port to run the kit on
+
+This means if the user runs more than one prototype, the first will be on port 3000, the next on 3001 and so on.
+
+- [Jest][jest] and [Cypress][cypress] to run tests
+
+We write unit and integration tests using Jest, and acceptance tests using Cypress.
+
+[expressjs]: https://expressjs.com/
+[nunjucks]: https://mozilla.github.io/nunjucks/
+[govuk-frontend]: https://frontend.design-system.service.gov.uk/
+[nodemon]: https://nodemon.io/
+[browsersync]: https://browsersync.io/
+[portscanner]: https://www.npmjs.com/package/portscanner
+[jest]: https://jestjs.io/
+[cypress]: https://www.cypress.io/
+
+## Working on the kit
+
+When you want to see what a user would when working on a prototype, the kit package in this repo has a number of npm scripts for developers:
+
+- `npm run start:package` - creates and starts a new prototype in a temporary folder
```

Comment:
```
node scripts/create-prototype-and-start
```
works better as it creates a live link to the Prototype Kit, I think @lfdebrux is going to create an npm script for it

Reply:
Script was update in #1884, I've updated these docs now.